analyses/copy_number_consensus_call/src/scripts/compare_variant_calling_updated.py

```diff
@@ -0,0 +1,257 @@
+    else:
+        with open(i) as file:
+
+            ## Strip the trailing spaces for each line
+            stripped_content = [i.rstrip() for i in file.readlines()]
+
+            ## Split each line up by any white space between columns
+            fin_input_content = [re.split('\s+',i) for i in stripped_content]
+
+            ## If the 1st column has '1' instead of 'chr1' for the chromosome number, add in 'chr'
+            if fin_input_content[0][0].find('chr') == -1:
+                for c,chromo in enumerate(fin_input_content):
+                    fin_input_content[c][0] = 'chr' + fin_input_content[c][0]
+
+    ## Add the content to a list
+    list_of_list += [fin_input_content]
+
+
+## Make a list of iteraby index for the MAIN for-loop below
+## This give a list as followed: [0,1,2]
+list_index = list(range(0,len(list_of_list)))
+
+
+## We have 3 items in list_of_list. Each item is the content of an input file
+## Make a list to store the final files
+fin_list =[]
+
+## Loop through list_index
+for j, jval in enumerate(list_index):
+    for k,kval in enumerate(list_index[j+1:]):
+
+        ## Compare 1st and 2nd file from list_of_list
+        ## Then compare the 1st and 3rd
+        ## Then compare the 2nd and 3rd
+        list1 = list_of_list[jval]
+        list2 = list_of_list[kval]
+
+        ## Define a variable to store the list of consensus CNVs
+        consensus_list = []
+
+        ## For every CNVs in list1, we look through the ENTIRETY of list2 and perform actions if 2 CNVs overlaps
+        ## Could probably speed this up by using a dictionary.
+        for m in list1:
+
+            ## Assign the start_pos, end_pos, and copy number of list1 to variables
+            start_list1 = int(m[1])
+            end_list1 = int(m[2])
+            cnv_list1 = m[3]
+
+            ## Make sure end > start
+            if start_list1 > end_list1:
+                start_list1, end_list1 = end_list1, start_list1
+
+            ## Assign variable for the CULMULATIVE coverage of the current CNV from list1
+            coverage_list1 = 0
+
+            ## Assign variables to store information for list2
+            list2_overlap_len = 0
+            list2_total_len = 0
+            list2_start_list = []
+            list2_end_list = []
+            list2_chr_str_end = ''
+
+
+            ## For every of the CNV in list one, we look over ALL CNVs in list 2 and make comparisons
+            for n in list2:
+
+                ## Assign start_pos, end_pos, and copy number of list2 to variables
+                start_list2 = int(n[1])
+                end_list2 = int(n[2])
+                cnv_list2 = n[3]
+
+                ## Make sure end > start
+                if start_list2 > end_list2:
+                    start_list2, end_list2 = end_list2, start_list2
+
+                ## For the current CNV in list1, if it overlaps with any CNV from list2
+                if m[0] == n[0] and start_list1 <= end_list2 and end_list1 >= start_list2:
```

Comment:
using the changes suggested above to make what is going on here more transparent.
```suggestion
                if chr_list1 == chr_list2 and start_list1 <= end_list2 and end_list1 >= start_list2:
```

Reply:
I agree! Thanks for the suggestion!